test-engineering/load/locustfiles/load.py

```diff
@@ -0,0 +1,60 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Load test shape module."""
+
+from typing import Type
+
+from locust import LoadTestShape, User
+from locustfile import ContileFirefoxUser, ContileNonFirefoxUser
+from pydantic import BaseModel
+
+
+class ShapeStage(BaseModel):
+    """Data defining a shape stage."""
+
+    run_time: int
+    users: int
+    spawn_rate: float
+    user_classes: list[Type[User]] = [ContileFirefoxUser, ContileNonFirefoxUser]
+
+
+class ContileLoadTestShape(LoadTestShape):
+    """A load test shape class for Contile (Duration: 10 minutes, Users: 200)."""
+
+    RUN_TIME: int = 600  # 10 minutes (must not be set to less than 1 minute)
+    USERS = 200
+
+    stages: list[ShapeStage]
+
+    def __init__(self):
+        super(LoadTestShape, self).__init__()
+
+        spawn_rate: float = round(self.USERS / 60, 2)
+        self.stages = [
+            # Stage 1: Spawn users in the first minute and dwell until the last minute
+            ShapeStage(
+                run_time=(self.RUN_TIME - 60),
+                users=self.USERS,
+                spawn_rate=spawn_rate,
+            ),
+            # Stage 2: Stop users in the last minute
+            ShapeStage(run_time=self.RUN_TIME, users=0, spawn_rate=spawn_rate),
+        ]
+
+    def tick(self) -> tuple[int, float, list[Type[User]] | None] | None:
```

Comment:
Nit: let's use a type alias to make this easier to read.

```
TickReturnType = tuple[int, float, list[Type[User]] | None]
def tick(self) -> TickReturnType | None:
    ...
```

Reply:
Change applied: https://github.com/mozilla-services/contile/pull/536/commits/8e6ca2b4266b6d43fe0ba380a9b63bbe9123f1b1